packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -291,7 +291,14 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
                 [self getVideoCompositionWithTransform:self->_preferredTransform
                                              withAsset:asset
                                         withVideoTrack:videoTrack];
-            item.videoComposition = videoComposition;
+            // Invalid values of video composition will throws an exception
+            // (https://github.com/flutter/flutter/issues/151031).
+            // When there is a problem with the parameters of video composition, set nil.
+            @try {
+              item.videoComposition = videoComposition;
+            } @catch (NSException *exception) {
```

Comment:
Unless it's impossible to know in advance that a value is invalid for some reason, creating an exception and then catching it is not correct behavior. Invalid arguments are a programming error, not a runtime error, so the plugin code should never be calling this with invalid arguments in the first place.

Reply:
In this plugin, invalid parameters may be caused by rederSize of videoCompositiom, but there are multiple cases in the documentation. However if we know that the parameter is abnormal, should we still let it be created successfully? I don't think setting videoComposition is necessary for the creation process.

Reply:
However, we should check the validity of videoComposition in this function. Still, we want to determine if the creation process should be interrupted.https://github.com/flutter/packages/blob/754de1918a339270b70971b6841cf1e04dd71050/packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m#L211

Reply:
> there are multiple cases in the documentation

Please provide a link to the documentation that you are referring to here, so that we are all working from the same information.

> However if we know that the parameter is abnormal, should we still let it be created successfully? I don't think setting invalid videoComposition is necessary for the creation process.

Please see my other unanswered question about whether the invalid values are coming from. Until there's an answer to that, there's no way to answer questions about how they should be handled. Bad values that come from plugin clients are very different from bad values that come from uncontrolled external video date, both of which are very different from bad values coming from a programming error in the plugin itself.